From a0391366d6a3c61045a2b8409a0714252f7180c9 Mon Sep 17 00:00:00 2001
From: Min Lin <linmin@eswincomputing.com>
Date: Mon, 14 Jul 2025 20:38:50 +0800
Subject: [PATCH 623/654] fix(fusb303b):mutex lock used before initialization

Changelogs:
1. In fusb303b.c, the mutext chip->logbuffer_lock was used in fusb303b_init(chip)
   before intialization in fusb303b_debugfs_init(chip).
   Fixed it by placing fusb303b_debugfs_init(chip) before fusb303b_init(chip)

Signed-off-by: Min Lin <linmin@eswincomputing.com>
---
 drivers/usb/typec/fusb303b.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/typec/fusb303b.c b/drivers/usb/typec/fusb303b.c
index 5ffc27640a05..b0dd7c49b2c0 100644
--- a/drivers/usb/typec/fusb303b.c
+++ b/drivers/usb/typec/fusb303b.c
@@ -568,8 +568,8 @@ static int fusb303b_probe(struct i2c_client *client)
 	chip->dev = &client->dev;
 	mutex_init(&chip->lock);
 	spin_lock_init(&chip->irq_lock);
-	fusb303b_init(chip);
 	fusb303b_debugfs_init(chip);
+	fusb303b_init(chip);
 	if (client->irq)
 	{
 		chip->gpio_int_n_irq = client->irq;
-- 
2.51.0

